spec/opal/core/string/gsub_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+describe 'String#gsub' do
+  it 'handles recursive gsub' do
+    PassSlotRx = /{(\d+)}/
```

Comment:
I think this ends up being a global constant, maybe it can be reused as a local var using lambdas:

```rb
pass_slot_rx = /{(\d+)}/
recurse_gsub = -> text {
  text.gsub(pass_slot_rx) {
    index = $1.to_i
    if index == 0
      recurse_gsub.call '{1}'
    else
      'value'
    end
  }
}
end
result = recurse_gsub.call '<code>{0}</code>'
result.should == '<code>value</code>'
```

Reply:
Done 😄